testing/tests/src/test/kotlin/ComponentCreatorFailureTest.kt

```diff
@@ -204,4 +204,59 @@ class ComponentCreatorFailureTest(
 
         compileRunAndValidate()
     }
+
+    @Test
+    fun `invalid sub-component factory methods`() {
```

Comment:
Maybe it would be easier for readers to mark errors in given source with comments.